src/client/components/ToggleSwitchInput.tsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import type { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+import type { Props } from '@guardian/source-react-components';
+import { css } from '@emotion/react';
+import {
+  neutral,
+  success,
+  textSans,
+  focusHalo,
+  transitions,
+  descriptionId,
+  generateSourceId,
+} from '@guardian/source-foundations';
+
+const inputStyles = css`
+  opacity: 0;
+  position: absolute;
+  top: 0;
+  left: 0;
+`;
+
+const labelStyles = css`
+  user-select: none;
+  position: relative;
+  ${textSans.small()};
+  display: flex;
+  color: ${neutral[7]};
+  align-items: center;
+  cursor: pointer;
+`;
+
+const switchStyles = (labelPosition: LabelPosition) => css`
+  flex: none;
+  border: none;
+  margin: ${labelPosition === 'left' ? '0px 0px 0px 8px' : '0px 8px 0px 0px'};
+  padding: 0;
+  display: inline-block;
+  text-align: center;
+  position: relative;
+  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
+  cursor: pointer;
+
+  &:after {
+    content: '';
+    position: absolute;
+    border-radius: 50%;
+    background: #fff;
+    will-change: left;
+    transition: left 0.15s ease-in-out;
+  }
+`;
+
+const siblingStyles = css`
+  input + span {
+    background-color: rgba(153, 153, 153, 0.5);
+  }
+
+  /* Required for Safari <15.3 */
+  input:focus + span {
+    transition: ${transitions.medium};
+    ${focusHalo};
+  }
+
+  /* When input receives focus that is not keyboard focus (eg. mouse) */
+  /* TODO does not work in chrome/edge/firefox if :focus is used for Safari*/
+  input:focus:not(:focus-visible) + span {
+    outline: none !important;
+  }
+
+  /* Focus-visible not fully supported in Safari <15.4 and IE 11 */
+  input:focus-visible + span {
+    transition: ${transitions.medium};
+    ${focusHalo};
+  }
```

Comment:
Couldn't find a solution that worked across all browsers so replicating the focus style behaviour of the check box which shows focus on mouse events still.

Reply:
Any better solutions let me know pls.

Reply:
Take a look at [https://guardian.github.io/source/?path=/story/packages-source-foundations-focusstylemanager--page](`FocusStyleManager`) which [Gateway is already using](https://github.com/guardian/gateway/blob/6653d8c3faa31a3ae743c74d7f337258a7c64c6a/src/client/static/sourceAccessibility.tsx)

Reply:
Thanks!